index.js

```diff
@@ -9,7 +9,17 @@ const pDebounce = (fn, wait, options = {}) => {
 	let timer;
 	let resolveList = [];
 
+	// IMPORTANT: Must preserve calling context that debounced fn is called
+	// * Returned debounced fn is written as `function() { }` so that its called `this` context
+	//   is known.
+	// * `new Promise(resolve => {})` and setTimeout's callbacks are arrow fn(), so `context` is
+	//   not required for ES6.  But if pDebounce is transpiled to ES5, `context` is required because
+	//   the arrow fn will become a `function() {...}`
+	// * `result` MUST be calculated by calling fn() in same context that pDebounce()'s returned
+	//   function is called.  Hence `fn.apply(context, arguments_)` rather than `fn(...arguments_)`
```

Comment:
I think this comment can be replaced by something like `Preserve the context when the debounced function is called on this`

Reply:
Thank you.  Simplifying the comment is helpful.  I added a commit that is similar to your suggestion.

Reply:
The comment is now removed given https://github.com/sindresorhus/p-debounce/pull/16#discussion_r561587514